src/mon/AuthMonitor.cc

```diff
@@ -1043,6 +1043,12 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
     goto done;
   }
 
+  //if capability strings are malformed, return with error msg.
+  if (!valid_caps(caps_vec, &ss)) {
+    err = -EINVAL;
+    goto done;
+  }
```

Comment:
Why did you shift this check earlier?

Reply:
It was suggested by @jecluis 
And I think it's better to check it right after the caps are generated, so in each specific cmd we don't need to check it repeatedly.

Reply:
Hmm, does this check pass if they're using eg "auth import" rather than a per-key command?